src/routes/_surfaces/workspace/explore/time-range-constructor/TimeRangeConstructor.svelte

```diff
@@ -0,0 +1,79 @@
+<!--
+@component
+Constructs a TimeRange object – to be used as the filter in MetricsExplorer – by taking as input:
+- the time range name (a semantic understanding of the time range, like "Last 6 Hours" or "Last 30 days")
+- the time grain (e.g., "hour" or "day")
+- the dataset's full time range (so its end time can be used in relative time ranges)
+-->
+<script lang="ts">
+  import type {
+    TimeGrain,
+    TimeRangeName,
+    TimeSeriesTimeRange,
+  } from "$common/database-service/DatabaseTimeSeriesActions";
+  import { setExploreSelectedTimeRangeAndUpdate } from "$lib/redux-store/explore/explore-apis";
+  import { getMetricsExplorerById } from "$lib/redux-store/explore/explore-readables";
+  import { store } from "$lib/redux-store/store-root";
+  import _ from "lodash";
+  import TimeGrainSelector from "./TimeGrainSelector.svelte";
+  import TimeRangeNameSelector from "./TimeRangeNameSelector.svelte";
+  import { makeTimeRange } from "./timeRangeUtils";
+
+  export let metricsDefId: string;
+
+  $: metricsExplorer = getMetricsExplorerById(metricsDefId);
+
+  let selectedTimeRangeName;
+  const setSelectedTimeRangeName = (timeRangeName: TimeRangeName) => {
+    selectedTimeRangeName = timeRangeName;
+  };
+
+  let selectedTimeGrain;
+  const setSelectedTimeGrain = (timeGrain: TimeGrain) => {
+    selectedTimeGrain = timeGrain;
+  };
+
+  const constructTimeRangeAndUpdateStore = (
+    timeRangeName: TimeRangeName,
+    timeGrain: TimeGrain,
+    allTimeRangeInDataset: TimeSeriesTimeRange
+  ) => {
+    if (!timeRangeName || !timeGrain || !allTimeRangeInDataset) return;
+
+    const newTimeRange = makeTimeRange(
+      selectedTimeRangeName,
+      selectedTimeGrain,
+      $metricsExplorer.allTimeRange
+    );
+
+    if (_.isEqual(newTimeRange, $metricsExplorer.selectedTimeRange)) return;
```

Comment:
Is thre a way we can do this without pulling in all of lodash? Aren't these time ranges just a few values to check?

Reply:
Yep, whoops, did that quickly. Just replaced with manual value comparisons.